modules/apps/web-experience/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java

```diff
@@ -308,17 +340,24 @@ protected void initPersonalControlPanelPortletPermission(
 
 			if (actionIds.contains(actionId)) {
 				resourcePermissionLocalService.addResourcePermission(
-					portlet.getCompanyId(), portlet.getRootPortletId(),
-					ResourceConstants.SCOPE_COMPANY,
-					String.valueOf(portlet.getCompanyId()),
+					companyId, portlet.getRootPortletId(),
+					ResourceConstants.SCOPE_COMPANY, String.valueOf(companyId),
 					userRole.getRoleId(), actionId);
 			}
 
 			portletLocalService.updatePortlet(
 				portlet.getCompanyId(), portlet.getPortletId(),
 				StringPool.BLANK, portlet.isActive());
+
+			portletPreferences.setValue(
+				_MY_ACCOUNT_PERMISSIONS_INITIALIZED, StringPool.TRUE);
+
+			portletPreferences.store();
 		}
 
+		private static final String _MY_ACCOUNT_PERMISSIONS_INITIALIZED =
+			"myAccountPermissionsInitialized";
+
```

Comment:
@brianchandotcom here is the larger context: https://github.com/JorgeFerrer/liferay-portal/pull/812